### DISCUSSION TITLE
Spy on function

### DISCUSSION THREAD

**Discussion by Rachel-Rabinowitz** [pytest-dev/pytest#10290] [Q&A]
I have something like this:

```python
def C():
     # some logic
     return something

def B():
     # some logic
     return C()

def A():
     b = B()
     # some logic
```

I want to test A and mock the return value of C
Is there any way to do that?

**Comment by nicoddemus** [comment]
Hi @Rachel-Rabinowitz,

Yes, you have to mock `C` in the module which contains `B`. Supposing they are all in the same module, `app.module`, you would go like this using pytest-mock:

```python
def test_a(mocker):
    mocker.patch("app.module.C", return_value=42)
    ...
```

**Reply by Rachel-Rabinowitz** [reply]
That is what I tried first but it didn't work,
the reason was that I used it via module,
I had an init.py with import *
probably the run was with full relative path

Thanks for the answere
